File: dotnet/eshop-cqrs-ddd-microservices-excercise/Com.Marekturis.Common/Application/Validation/ValidationException.cs
using System;
using System.Runtime.Serialization;

namespace Com.Marekturis.Common.Application.Validation
{
    public class ValidationException : Exception
    {
        public string ErrorMessage => Message;

        public ValidationException(string message) : base(message)
        {
        }

        public ValidationException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}

File: java/eshop-cqrs-ddd-microservices-excercise/common/src/main/java/com/marekturis/common/application/validation/ValidationException.java
package com.marekturis.common.application.validation;

/**
 * @author Marek Turis
 */
public class ValidationException extends RuntimeException {

	public ValidationException(String message) {
		super(message);
	}

	public ValidationException(String message, Throwable cause) {
		super(message, cause);
	}

	public String getValidationErrorMessage() {
		return getMessage();
	}
}
